runbotics/runbotics-orchestrator-ui/src/src-app/views/process/EditProcessDialog.tsx

```diff
@@ -1,99 +1,164 @@
-import React, { FC, useState } from 'react';
+import React, { ChangeEvent, FC, useState, useEffect, KeyboardEvent } from 'react';
 
 import {
-    Dialog,
-    DialogActions,
-    DialogContent,
-    DialogTitle,
-    Button,
-    FormControl,
-    TextField,
-    Switch,
-    FormControlLabel,
-    Box,
+    DialogActions, DialogContent, DialogTitle, Dialog,
+    Button, TextField, Autocomplete,
 } from '@mui/material';
-import { IProcess } from 'runbotics-common';
 
+import { unwrapResult } from '@reduxjs/toolkit';
+import { IProcess, Tag } from 'runbotics-common';
 
+import useDebounce from '#src-app/hooks/useDebounce';
 import useTranslations from '#src-app/hooks/useTranslations';
+import { useDispatch } from '#src-app/store';
+import { processActions } from '#src-app/store/slices/Process';
 
-import BotCollectionComponent from './ProcessConfigureView/BotCollection.component';
-import BotSystemComponent from './ProcessConfigureView/BotSystem.component';
+import { EditProcessDialogSelectFields, EditProcessDialogTextFields } from './EditProcessDialogForm';
 import { Form } from '../utils/FormDialog.styles';
 
+const DEBOUNCE_TIME = 300;
+const MAX_NUMBER_OF_TAGS = 15;
+const MAX_TAG_LENGTH = 20;
 
-type EditProcessDialogProps = {
+interface EditProcessDialogProps {
     open?: boolean;
     process: IProcess;
     onClose: () => void;
     onAdd: (process: IProcess) => void;
 };
 
+export interface FormValidationState {
+    name: boolean;
+}
+
+const initialFormValidationState: FormValidationState = {
+    name: true
+};
+
 const EditProcessDialog: FC<EditProcessDialogProps> = ({
     process, onAdd, onClose, open,
 }) => {
-    const [formState, setFormState] = useState<IProcess>({ ...process });
     const { translate } = useTranslations();
+    const dispatch = useDispatch();
 
-    const handleSubmit = () => {
-        onAdd(formState);
+    const [formValidationState, setFormValidationState] = useState<FormValidationState>(initialFormValidationState);
+    const [processFormState, setProcessFormState] = useState<IProcess>({ ...process });
+    const [searchedDatabaseTags, setSearchedDatabaseTags] = useState<Tag[]>([]);
```

Comment:
what does it do? looks weird

Reply:
That was state names changing after Stanisław review

Reply:
If i understand well these are the tags that already exist in the database but not all of them - only the ones that match what user types while adding the tag: 
<img width="301" alt="image" src="https://github.com/runbotics/runbotics/assets/69820069/011c343d-0451-43d7-a29f-033d5230345f">